packages/analytics/dashboard-renderer/src/components/layout/GridLayout.vue

```diff
@@ -0,0 +1,130 @@
+<template>
+  <div
+    ref="gridContainer"
+    class="kong-ui-public-grid-layout"
+  >
+    <div
+      v-for="cell in totalCells"
+      :key="cell.key"
+      class="grid-cell"
+      :class="{
+        'empty-cell': !cell.tile,
+      }"
+      :style="cell.tile && {
+        gridRow: `span ${cell.tile.size.rows}`,
+        gridColumn: `span ${cell.tile.size.cols}`,
+      }"
+    >
+      <slot
+        v-if="cell.tile"
+        class="tile-content"
+        name="tile"
+        :tile="cell.tile"
+      />
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { computed, type PropType, ref, reactive } from 'vue'
+import type { GridSize, Tile, Cell } from 'src/types'
+import { DEFAULT_TILE_HEIGHT, DEFAULT_TILE_WIDTH } from '../../constants'
+
+const props = defineProps({
+  gridSize: {
+    type: Object as PropType<GridSize>,
+    required: true,
+  },
+  tileHeight: {
+    type: Number,
+    required: false,
+    default: DEFAULT_TILE_HEIGHT,
+  },
+  tileWidth: {
+    type: Number,
+    required: false,
+    default: DEFAULT_TILE_WIDTH,
+  },
+  tiles: {
+    type: Array as PropType<Tile[]>,
+    required: true,
+  },
+})
+
+const gridCols = ref(props.gridSize.cols)
+const gridRows = ref(props.gridSize.rows)
+const gridContainer = ref<HTMLElement>()
```

Comment:
issue: you have to initialize the ref as `null`, not `undefined`
```suggestion
const gridContainer = ref<HTMLElement>(null)
```